pkg/task/restore.go

```diff
@@ -599,3 +603,37 @@ func restoreTableStream(
 		}
 	}
 }
+
+func filterExistsSchema(
+	dom *domain.Domain,
+	dbs []*utils.Database,
+	tables []*utils.Table) (
+	[]*utils.Database, []*utils.Table) {
+	infoSchema := dom.InfoSchema()
+
+	existsDBs := make([]string, 0, len(dbs))
+	nonExistsDBs := make([]*utils.Database, 0, len(dbs))
+	for _, db := range dbs {
+		if ok := infoSchema.SchemaExists(db.Info.Name); !ok {
+			nonExistsDBs = append(nonExistsDBs, db)
+		} else {
+			existsDBs = append(existsDBs, db.Info.Name.String())
+		}
+	}
+
+	existsTables := make([]string, 0, len(dbs))
+	nonExistsTables := make([]*utils.Table, 0, len(tables))
+	for _, table := range tables {
+		if ok := infoSchema.TableExists(table.Db.Name, table.Info.Name); !ok {
+			nonExistsTables = append(nonExistsTables, table)
```

Comment:
We filtered out the table if it exists, but `GoValidateFileRanges` leads to that we won't restore files associated to tables if the table absent at `tableStream`... (i.e. The table won't join the pipeline, hence won't be restored...)

Reply:
ok, maybe we should send table to `tableStream` even table already exists